org.eclipse.xtext/src/org/eclipse/xtext/nodemodel/impl/NodeModelBuilder.java

```diff
@@ -273,16 +273,17 @@ protected void replaceChildren(AbstractNode oldNode, AbstractNode newNode) {
 		}
 	}
 	
-	public void replaceAndTransferLookAhead(INode oldNode, INode newRootNode) {
+	private AbstractNode replace(INode oldNode, INode newRootNode) {
 		AbstractNode newNode = ((CompositeNode) newRootNode).basicGetFirstChild();
 		replaceWithoutChildren((AbstractNode) oldNode, newNode);
 		if (oldNode instanceof ICompositeNode && newNode instanceof CompositeNode) {
 			CompositeNode newCompositeNode = (CompositeNode) newNode;
 			newCompositeNode.basicSetLookAhead(((ICompositeNode) oldNode).getLookAhead());
```

Comment:
Here is where the lookAhead is transfered.

Reply:
Thanks. I have renamed the methods